src/components/protocol_handler/test/protocol_payload_test.cc

```diff
@@ -258,8 +258,9 @@ TEST(ProtocolPayloadTest, ExtractProtocolWithJSONWithDataWithWrongPayloadSize) {
   prot_payload_test.json = expect_output_json_string;
   prot_payload_test.header.json_size = prot_payload_test.json.length();
 
-  const size_t data_for_sending_size =
-      PROTOCOL_HEADER_V2_SIZE + prot_payload_test.json.length();
+  const size_t data_for_sending_size = PROTOCOL_HEADER_V2_SIZE +
+                                       prot_payload_test.json.length() + 
+                                       prot_payload_test.data.size();
```

Comment:
@VENGEL Please confirm here why add ```data.size()```. I notice it not happens on SmartDeviceLink Repo